lib/mongo/server/connection_pool.rb

```diff
@@ -21,31 +21,45 @@ class Server
     #
     # @since 2.0.0
     class ConnectionPool
+      include Id
       include Loggable
+      include Monitoring::Publishable
       extend Forwardable
 
       # Create the new connection pool.
       #
       # @example Create the new connection pool.
-      #   Pool.new(wait_queue_timeout: 0.5) do
+      #   Pool.new(server, wait_queue_timeout: 0.5) do
       #     Connection.new
       #   end
       #
       # @note A block must be passed to set up the connections on initialization.
       #
-      # @param [ Hash ] options The connection pool options.
-      #
       # @option options [ Integer ] :max_pool_size The maximum pool size.
       # @option options [ Integer ] :min_pool_size The minimum pool size.
       # @option options [ Float ] :wait_queue_timeout The time to wait, in
       #   seconds, for a free connection.
       #
       # @since 2.0.0
-      def initialize(options = {}, &block)
-        @options = options.freeze
-        @queue = Queue.new(options, &block)
+      # @api private
+      def initialize(server, &block)
+        @id = ConnectionPool.next_id
+        @address = server.address.to_s
```

Comment:
Keep the address as a `Mongo::Address` object and expose it as such in all events, for consistency with the rest of the driver.